src/core/Service.js

```diff
@@ -73,6 +75,11 @@ export default class Service extends EventEmitter {
       type: this.serviceType,
       args: request
     };
+
+    if (timeout !== undefined) {
+      call.timeout = timeout;
+    }
+
```

Comment:
Is there a reason that this can't be put in the definition of the object above? Does passing `timeout: undefined` into rosbridge break it? That seems like an API issue, since it can be tricky to easily discern in a JavaScript application whether something is _deliberately undefined_ or _missing_.

I'd rather this read
```suggestion
      args: request,
      timeout: timeout
    };
```

and if this causes an issue in rosbridge we should fix it there.

Reply:
It seems that properties with undefined value are just being omitted when converted to JSON.